bootstrap/bootstrap-kubestellar.sh

```diff
@@ -194,67 +248,131 @@ if [ "$folder" == "" ]; then
+    if [ "$(kcp_version)" != "$KCP_REQUIRED_VERSION" ]; then
+        echo "kcp version $(kcp_version) is not supported, KubeStellar requires kcp $KCP_REQUIRED_VERSION ... exiting!"
+        exit 4
+    else
+        echo "kcp version $(kcp_version) ... ok"
+    fi
+    echo "Export KUBECONFIG environment variable: export KUBECONFIG=\"$KUBECONFIG\""
+    user_exports="$user_exports"$'\n'"export KUBECONFIG=\"$KUBECONFIG\""
 fi
 
 # Ensure KubeStellar is installed
-if [ "$(kubestellar_installed)" == "false" ]; then
-    if [ "$verbose" != "" ]; then
-        echo "Installing Kubestellar..."
+echo "************************************************"
+echo "Ensure KubeStellar is installed"
+echo "************************************************"
+if [ "$(kubestellar_installed)" == "true" ]; then
+    echo "KubeStellar found in the PATH at '$(which kubestellar)' ... skip installation."
+else
+    echo "KubeStellar not found in the PATH."
+    if [ "$(kubestellar_running)" == "true" ]; then
+        echo "KubeStellar processes are running already ... add KubeStellar folder to the PATH or stop KubeStellar with \"kubestellar stop\" ... exiting."
+        exit 5
     fi
-    bash <(curl -s https://raw.githubusercontent.com/kcp-dev/edge-mc/main/bootstrap/install-kubestellar.sh) --version $kubestellar_version --os $os_type --arch $arch_type --ensure-folder  $folder/kubestellar $verbose $flagx
-    if [[ ! ":$PATH:" == *":$(get_full_path $folder/kubestellar/bin):"* ]]; then
-        export PATH=$PATH:$(get_full_path $folder/kubestellar/bin)
+    ensure_folder "$folder/kubestellar"
+    kubestellar_folder=$(get_full_path "$folder/kubestellar")
+    kubestellar_bin_folder="$kubestellar_folder/bin"
+    echo "Downloading KubeStellar $kubestellar_version $os_type/$arch_type..."
+    kubestellar_download
+    echo "Installing KubeStellar into '$kubestellar_folder'..."
+    kubestellar_install
+    if [[ ! ":$PATH:" == *":$kubestellar_bin_folder:"* ]]; then
+        export PATH=$kubestellar_bin_folder:$PATH
+        echo "Add KubeStellar folder to the PATH: export PATH=\"$kubestellar_bin_folder:\$PATH\""
+        user_exports="$user_exports"$'\n'"export PATH=\"$kubestellar_bin_folder:\$PATH\""
     fi
 fi
 
-kubectl ws root
-
 # Ensure KubeStellar is running
-if [ "$(kubestellar_running)" == "false" ] || ! kubectl get workspaces.tenancy.kcp.io espw &> /dev/null ; then
-    if [ "$verbose" != "" ]; then
-        echo "Starting or restarting Kubestellar..."
+echo "************************************************"
+echo "Ensure KubeStellar is running"
+echo "************************************************"
+kubectl ws root > /dev/null
+if [ "$(kubestellar_running)" == "true" ]; then
+    echo "KubeStellar processes are running ... ok"
+    if ! kubectl get workspaces.tenancy.kcp.io espw &> /dev/null ; then
+        echo "KubeStellar ESPW does not exists ... run 'kubestellar stop' first ... exiting!"
```

Comment:
Was it a considered choice to not do `kubestellar start` in this case?  `kubestellar start` would stop all the running controllers, ensure the ESPW exists, and start all the controllers.

Reply:
I decided to exit with an error to let the user decide what to do.
I think the cleaner solution could be a `kubestellar stop` && `kubesterllar start`.

Reply:
So it is a considered choice.  Good enough for me.